kaiax/gov/headergov/impl/init.go

```diff
@@ -101,6 +122,28 @@ func (h *headerGovModule) PopMyVotes(idx int) {
 	h.myVotes = append(h.myVotes[:idx], h.myVotes[idx+1:]...)
 }
 
+// scanAllVotesInHeader scans all votes from headers in the given epoch.
+func (h *headerGovModule) scanAllVotesInHeader(epochIdx uint64) map[uint64]headergov.VoteData {
+	rangeStart := epochIdx * h.epoch
+	rangeEnd := (epochIdx + 1) * h.epoch
+
+	votes := make(map[uint64]headergov.VoteData)
+	for blockNum := rangeStart; blockNum < rangeEnd; blockNum++ {
+		header := h.Chain.GetHeaderByNumber(blockNum)
+		vote, err := headergov.VoteBytes(header.Vote).ToVoteData()
+		if err != nil {
+			logger.Error("Failed to parse vote", "num", blockNum, "err", err)
+			continue
+		}
+		// TODO-kaiax: consider writing addval/removeval votes to validator DB.
```

Comment:
If we're going to separate gov and valset schemas, then we can ignore addvalidator/removevalidator votes here as well.

Reply:
Then scanning the header will cost twice as much (two DB reads for each header).
Maybe we can come up with an optimization later.

Reply:
Sure. Let's come up with optimization later.